physics/src/modules/OceanBoundaryModule/TOPAZOcean.cpp

```diff
@@ -97,9 +120,6 @@ void TOPAZOcean::setData(const ModelState::DataMap& ms)
     slabOcean.setData(ms);
 }
 
-void TOPAZOcean::updateTf(size_t i, const TimestepTime& tst)
-{
-    tf[i] = Module::getImplementation<IFreezingPoint>()(sss[i]);
-}
+void TOPAZOcean::updateTf(size_t i, const TimestepTime& tst) { tf[i] = (*pFreezingPoint)(sss[i]); }
```

Comment:
I'd rather this continued to use the `Module` infrastructure.

Reply:
Ah, yes. I should have asked you about this one. See commit message for bb47c1e1

> I don't understand why, but using Module::getImplementation<Interface>
> directly reset nOcean, oceanIndex, and p_oceanMaskH. But if I first get
> a pointer to the implementation everything works. A bonus is that we can
> call tryConfigure for these modules in a transparent way.